src/db.c

```diff
@@ -149,10 +149,11 @@ robj *lookupKeyReadWithFlags(redisDb *db, robj *key, int flags) {
     return lookupKey(db, key, flags);
 }
 
-/* Like lookupKeyReadWithFlags(), but does not use any flag, which is the
+/* Like lookupKeyReadWithFlags(), but sets flags automatically, which is the
  * common case. */
-robj *lookupKeyRead(redisDb *db, robj *key) {
-    return lookupKeyReadWithFlags(db,key,LOOKUP_NONE);
+robj *lookupKeyRead(client *c, robj *key) {
+    int flags = c->flags & CLIENT_STEALTHMODE ? LOOKUP_NOTOUCH : LOOKUP_NONE;
+    return lookupKeyReadWithFlags(c->db, key, flags);
```

Comment:
Interesting approach that lookupKeyRead takes a client but lookupKeyReadWithFlags doesn't.

Maybe the names of these functions should be changed, since the "WithFlags" version was the same as the other one, except it also takes flags. Now there is one more difference that perhaps can be reflected in the names somehow.

This change also creates an assymetry between lookupKeyRead and lookupKeyWrite. Maybe lookupKeyWrite should also take a client, for the symmetry?

Reply:
I agree, I think all of the lookupKey should take in the client structure.

Reply:
Perhaps `lookupKeyReadDefaultFlags` and `lookupKeyReadCustomFlags`?

Reply:
Sounds good, but do we always have a client?

Reply:
The only case we don't have a client is in `blocked.c/handleClientsBlockedOnKeys` ... but in this case we need to search all blocked clients to get the client we need.

Reply:
Another problem about symmetry: `lookupKeyWrite` in `copyCommand` and `moveCommand` have to take in db, as the destination db is not contained in client structure. I am considering two approaches: 

- All lookupKey functions can take in both client and db, and look up keys in db, use client flags if needed.
- Those original lookupKey functions still takes in db, we can have a new function `lookupKeyReadStealth`, which takes in client. We don't need a `lookupKeyWriteStealth`, and I don't think it breaks the symmetry so much, since read and write operations are just different.

Personally I prefer the second approach. @zuiderkwast @madolson WDYT?